File: extract_pdfs.py
import os
from pypdf import PdfReader

def extract_text(pdf_path, txt_path):
    print(f"Extracting {pdf_path}...")
    try:
        reader = PdfReader(pdf_path)
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        
        with open(txt_path, "w", encoding="utf-8") as f:
            f.write(text)
        print(f"Saved to {txt_path}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    mat_dir = r"D:\Quiz\Material"
    pdf1 = os.path.join(mat_dir, "POLÍTICA DE SEGURANÇA DA INFORMAÇÃO - PSI.pdf")
    txt1 = os.path.join(mat_dir, "psi_text.txt")
    
    pdf2 = os.path.join(mat_dir, "POLÍTICA DE proteção e privacidade de dados - PPDP.pdf")
    txt2 = os.path.join(mat_dir, "ppdp_text.txt")
    
    extract_text(pdf1, txt1)
    extract_text(pdf2, txt2)

File: backend/extract_pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extract(path, outFile) {
    console.log('Extracting', path);
    try {
        let dataBuffer = fs.readFileSync(path);
        let data = await pdf(dataBuffer);
        fs.writeFileSync(outFile, data.text);
        console.log('Saved to', outFile);
    } catch (e) {
        console.error('Error extracting', path, e);
    }
}

async function run() {
    await extract(
        'D:\\Quiz\\Material\\POLÍTICA DE SEGURANÇA DA INFORMAÇÃO - PSI.pdf',
        'D:\\Quiz\\Material\\psi_text.txt'
    );
    await extract(
        'D:\\Quiz\\Material\\POLÍTICA DE proteção e privacidade de dados - PPDP.pdf',
        'D:\\Quiz\\Material\\ppdp_text.txt'
    );
}

run();
